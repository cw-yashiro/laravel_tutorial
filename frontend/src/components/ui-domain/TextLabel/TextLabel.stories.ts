import { TextLabel } from "./TextLabel"

export default {
  title: "ui-domain/TextLabel",
  component: TextLabel,
}

export const Default = {
  args: {
    label: "テキストラベル",
  },
}
