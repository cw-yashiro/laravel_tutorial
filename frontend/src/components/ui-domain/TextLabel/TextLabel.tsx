import { forwardRef } from "react"

type Props = React.ComponentPropsWithRef<"label">

export const TextLabel = forwardRef(({ ...props }: Props) => {
  return <label {...props} />
})
