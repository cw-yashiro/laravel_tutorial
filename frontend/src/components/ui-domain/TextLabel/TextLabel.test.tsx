import { describe, it, expect } from "vitest"
import { render } from "@testing-library/react"
import { TextLabel } from "./TextLabel"

describe("ui-domain/TextLabel", () => {
  it("Snapshot: テキストラベルが表示されている", () => {
    const { container } = render(<TextLabel>テキストラベル</TextLabel>)
    expect(container).toMatchSnapshot()
  })
})
