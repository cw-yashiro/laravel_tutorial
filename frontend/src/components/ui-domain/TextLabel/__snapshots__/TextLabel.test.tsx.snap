// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui-domain/TextLabel > Snapshot: テキストラベルが表示されている 1`] = `
<div>
  <label>
    テキストラベル
  </label>
</div>
`;
